synapse/push/bulk_push_rule_evaluator.py

```diff
@@ -274,7 +274,7 @@ def _condition_checker(
     cache: Dict[str, bool],
 ) -> bool:
     for cond in conditions:
-        _id = cond.get("_id", None)
+        _id = cond.get("_cache_key", None)
         if _id:
             res = cache.get(_id, None)
```

Comment:
Probably worth making the `_id` variable into `_cache_key` also?

Reply:
https://github.com/matrix-org/synapse/pull/12188/commits/3e8dac0dcc25ddedcfff7c6dcdb457599d038f57